integrations/freshchat/src/actions/hitl.ts

```diff
@@ -0,0 +1,173 @@
+import { RuntimeError } from '@botpress/client'
+import { getFreshchatClient } from 'src/client'
+import * as bp from '.botpress'
+
+export const startHitl: bp.IntegrationProps['actions']['startHitl'] = async ({ ctx, client, input, logger }) => {
+  try {
+    const freshchatClient = getFreshchatClient({ ...ctx.configuration }, logger)
+
+    const { user } = await client.getUser({
+      // Retrieve the user ID created in the createUser action
+      id: input.userId,
+    })
+
+    if (!user.tags.id?.length) {
+      throw new RuntimeError("Input user doesn't have a Freshchat User Id")
+    }
+
+    const { title, description, messageHistory } = input
+
+    const {
+      state: {
+        payload: { channelId },
+      },
+    } = await client.getState({
+      type: 'integration',
+      id: ctx.integrationId,
+      name: 'freshchat',
+    })
+
+    const messages: any[] = [
+      {
+        message_parts: [
+          {
+            text: {
+              content: `New Conversation Started
+
+              Title: ${title}
+              Description: ${description}
+            `,
+            },
+          },
+        ],
+        channel_id: channelId,
+        message_type: 'normal',
+        actor_type: 'user',
+        actor_id: user.tags.id,
+      },
+    ]
+
+    messages.push({
+      message_parts: [
+        {
+          text: {
+            content: `Transcript:
+            ${
+              messageHistory
+                ?.map((message) => {
+                  let text = ''
+
+                  if (message.type !== 'text') {
+                    text = `(Event: ${message.type})`
+                  } else {
+                    text = message.payload.text
+                  }
+
+                  const origin =
+                    message.source.type === 'bot' ? 'Bot: ' : message.source.userId === user.id ? 'User: ' : ''
+
+                  return `${origin}${text}`
+                })
+                .join('\n') || '-'
+            }
+          `,
+          },
+        },
+      ],
+      channel_id: channelId,
+      message_type: 'normal',
+      actor_type: 'user',
+      actor_id: user.tags.id,
+    })
+
+    const freshchatConversation = await freshchatClient.createConversation({
+      userId: user.tags.id as string,
+      messages,
+      channelId,
+    })
+
+    const { conversation } = await client.getOrCreateConversation({
+      channel: 'hitl',
+      tags: {
+        id: `${freshchatConversation.conversation_id}`,
+      },
+    })
+
+    return {
+      conversationId: conversation.id,
+    }
+  } catch (error: any) {
```

Comment:
**nitpick:** use unknown

Reply:
using unknown generates typing errors

Reply:
yes that's the goal; here's how to do it the correct way:
```ts
} catch (thrown: unknown) {
  const err = thrown instanceof Error ? thrown : new Error(String(thrown))
  // your error is typed and there is no risk of throwing inside the catch block (this would be very very bad)
}
```

Reply:
Got your point, typing is the way forward, not sure what I feel about the additional code for readability, I will revisit this in future versions to check if there are better ways type it